### PR TITLE
Makes omnisentries act on battery life as well as makes them rotate

### PR DESCRIPTION
**Pull request by NewyearnewmeUwu**
# About the pull request
 
This PR nerfs omnisentries down to 500 rounds per sentry and makes them rotate. Each time they rotate, they have a delay, so that faster castes can outmaneuver it. Thanks to @carlarctg for da help with rotating.

It also adds a timed life for the sentries so that they are emphasized as _temporary_ field assets. Each shot reduces the battery life (10 minutes) by 3 seconds as of writing, and you can save battery by turning the sentry off. As they say, use it or lose it.

# Explain why it's good for the game
Omnisentries are a pain to fight for most castes because of the fact that they're usually hidden in 1x1 wired cade boxes. They also have an _absurd_ 1500 ammo pool, which makes no sense. Why are airdroppable sentries carrying 3x more than the default ones comtechs use? Plus, it looks really cool, and emphasizes their role as positional harassment vs instant death laser pillboxes that you need a T3 to deal with. The limited lifespan also incentivizes smart placement of them versus simply using them as the previously-stated instant death laser pillboxes, while providing a means of counterplay to xenos- each shot takes 5 seconds off the battery life.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
balance: omnisentries now rotate, have 500 rounds total, and have a 5 minute lifespan. Each shot reduces that lifespan by 5 seconds.
/:cl:

### CODE REVIEW

**Comment by cuberound** [comment]
if you want it to be used in certain way, what about instead of making it worse overall in every role. honestly, omnisentry SUCKS as position harassment, when single T3 can just walk to them and destroy them, when they are not protected by cades, eaven when there are marines around. it is not gona be used as frontline harassment with 4 tile range and now with rotate super long cooldown that makes anyone able to outmanuver them.

File: code/modules/defenses/sentry.dm [hunk 1]
```diff
@@ -107,6 +112,8 @@
 
 /obj/structure/machinery/defenses/sentry/get_examine_text(mob/user)
 	. = ..()
+	if(has_lifespan == TRUE)
```
Comment (mullenpaul) [inline code/modules/defenses/sentry.dm:115]: Same as above. This examine text can also be put into the subclasses get_examine_text

File: code/modules/defenses/sentry.dm [hunk 2]
```diff
@@ -217,15 +224,19 @@
...
+		playsound(loc, 'sound/mecha/mechturn.ogg', 50, 1)
+		last_fired = world.time + 0.5 SECONDS
+		return
 	actual_fire(A)
 
 	if(targets.len)
 		addtimer(CALLBACK(src, PROC_REF(get_target)), fire_delay)
 
 /obj/structure/machinery/defenses/sentry/proc/actual_fire(var/atom/A)
+	if (has_lifespan == TRUE)
```
Comment (mullenpaul) [inline code/modules/defenses/sentry.dm:238]: Same as above. I would also put this into the subclass

**Review by mullenpaul — CHANGES_REQUESTED** [review]
I've reviewed the code as presented but I'm not convinced that a 'battery life' is a good way of balancing this. If a battery charge is the path forward then this code should be reworked to use a battery, rather than a timer with an IC justification of an internal power store. It seems pretty gamey to put a battery life on this sentry when literally no other sentry does.

**Pull request closed without merging**